tests/test_lists2safebrowsing.py

```diff
@@ -168,3 +215,279 @@ def test_add_domain_to_list_duplicate():
+            with patch("test_lists2safebrowsing.open", mock_open()):
+                with open("test-list-digest256", "wb") as output_file:
+                    log_file = log_file if log else None
+                    function(incoming, chunknum, output_file, log_file,
+                             section)
+
+    log_writes = log_file.write.call_args_list if log else []
+    output_writes = output_file.write.call_args_list
+
+    return (output_writes, log_writes, expected_output_writes,
+            expected_log_writes, expected_print)
+
+
+@pytest.mark.parametrize("list_type", [LIST_TYPE_ENTITY, LIST_TYPE_PLUGIN])
+def test_process_list(capsys, chunknum, list_type):
+    """Validate entity/plugin list generation."""
+    (output_writes, log_writes, expected_output_writes, expected_log_writes,
+     expected_print) = _process_entity_or_plugin_list(chunknum, list_type)
+
+    assert output_writes == expected_output_writes
+    assert log_writes == expected_log_writes
+    assert capsys.readouterr().out == expected_print
+
+
+@pytest.mark.parametrize("list_type", [LIST_TYPE_ENTITY, LIST_TYPE_PLUGIN])
+def test_process_list_no_log_file(capsys, chunknum, list_type):
+    """Test entity/plugin list generation without a log file."""
+    (output_writes, _, expected_output_writes, _, expected_print) = \
+        _process_entity_or_plugin_list(chunknum, list_type, log=False)
+
+    assert output_writes == expected_output_writes
+    assert capsys.readouterr().out == expected_print
+
+
+def _get_entity_or_plugin_lists(list_type, config, section, chunknum,
+                                data, domains):
+    """Auxiliary function for get_entity_lists/get_plugin_lists tests."""
+    if list_type == LIST_TYPE_ENTITY:
+        function = l2s.get_entity_lists
+        url = ENTITY_LIST_URL
+    elif list_type == LIST_TYPE_PLUGIN:
+        function = l2s.get_plugin_lists
+        url = PLUGIN_LIST_URL
+    else:
+        raise ValueError("Invalid list type: '%s'" % list_type)
+
+    with patch("lists2safebrowsing.urllib2.urlopen",
+               mock_open(read_data=data)) as mocked_urlopen, \
+            patch("lists2safebrowsing.open", mock_open()) as mocked_open:
+        output_file, _ = function(config, section, chunknum)
+
+    urlopen_calls = mocked_urlopen.call_args_list
+    expected_urlopen_calls = [call(url)]
+    open_calls = mocked_open.call_args_list
+    output_filename = config.get(section, "output")
+    expected_open_calls = [call(output_filename, "wb"),
+                           call(output_filename + ".log", "w")]
+
+    # Exclude log writes
+    output_writes = output_file.write.call_args_list[len(domains):]
+    expected_output_writes = _get_expected_results(list_type, domains,
+                                                   chunknum, section)[0]
+
+    return (urlopen_calls, expected_urlopen_calls, open_calls,
+            expected_open_calls, output_writes, expected_output_writes)
```

Comment:
As I mentioned in ` _get_expected_results` and in `_write_safebrowsing_blocklist` can you call the function as it should on the unittests and compare the expected values against static values? Testing to make sure that the section corresponds to the correct function being called should happen on a different unittest for main. For now, it's good enough to use the specific sections that we know the `get_entity_lists` or `get_plugin_lists` will be called with.